tasks/au/meta/helper2016/au_columns_helper.py

```diff
@@ -0,0 +1,205 @@
+           and not short.startswith('M_') and not short.endswith('_M') \
+           and not short.startswith('F_') and not short.endswith('_F')
+
+
+def _filter_denominators(denominators, datapack):
+    filtered = denominators
+    for filter_ in DENOMINATOR_FILTERS:
+        filtered = list(filter(lambda x: filter_ not in x, filtered))
+    filtered = [DENOMINATOR_ALIASES['all'].get(x, x) for x in filtered]
+    filtered = [DENOMINATOR_ALIASES.get(datapack, {}).get(x, x) for x in filtered]
+
+    return filtered
+
+
+with open(os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                       'au_metadata_columns.csv'), 'r') as infile:
+    with open(os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                           'Metadata_2016_GCP_DataPack.csv'), 'a') as outfile:
+        with tempfile.NamedTemporaryFile(mode='w+', delete=False) as tempfile_:
+            reader = csv.reader(infile, delimiter=',', quotechar='"')
+            tempwriter = csv.writer(tempfile_, delimiter=',', quotechar='"', quoting=csv.QUOTE_MINIMAL)
+            writer = csv.writer(outfile, delimiter=',', quotechar='"', quoting=csv.QUOTE_MINIMAL)
+            writer.writerow(['Sequential', 'Short', 'Name', 'DENOMINATORS', 'Tablename', 'unit', 'subsection',
+                            'Column heading description in profile', 'AGG', 'order', 'Table descr'])
+            for row in reader:
+                table = row[3][:3]
+
+                if table not in tables.keys():
+                    continue
+
+                sequential = row[0]
+                short = row[1]
+                name = row[2].replace('_', ' ')
+                datapack = row[3]
+                profile_table = row[4]
+                description = row[5]
+                denominators = []
+
+                if short.startswith('P_') or short.endswith('_P') and short != 'Tot_P_P':
+                    denominators.append('Tot_P_P')
+                elif short.startswith('M_'):
+                    denominators.append('Tot_P_P')
+                    denominators.append('Tot_P_M')
+                    denominators.append('P_' + short[2:])
+                elif short.endswith('_M'):
+                    denominators.append('Tot_P_P')
+                    if short != 'Tot_P_M':
+                        denominators.append('Tot_P_M')
+                    denominators.append(short[:short.rfind('_M')] + '_P')
+                elif short.startswith('F_'):
+                    denominators.append('Tot_P_P')
+                    denominators.append('Tot_P_F')
+                    denominators.append('P_' + short[2:])
+                elif short.endswith('_F'):
+                    denominators.append('Tot_P_P')
+                    if short != 'Tot_P_F':
+                        denominators.append('Tot_P_F')
+                    denominators.append(short[:short.rfind('_F')] + '_P')
+
+                denominators = _filter_denominators(denominators, datapack)
```

Comment:
Defensive programming suggestion, feel free to ignore: as we can't have valuable tests for this kind of stuff, maybe we could check that all rows have the same number of columns, and fail otherwise.